src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Models/LocalConfigurationClient.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Azure.Data.AppConfiguration;
+using System;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Models
+{
+    internal class LocalConfigurationClient
```

Comment:
The PR adds IConfigurationClient, this doesn't implement that. Seems off.

Reply:
I agree, it is little misleading.
The reason for the `LocalConfigurationClient` to exist is because we needed a mechanism with which we could -
1. Have an ordered list of ConfigurationClients that we can iterate through. (in `FailOverSupportedConfigurationClient`)
2. Having a mapping of Endpoint -> ConfigurationClient.

The existing solutions like `OrderedDictionary` were not well suited for this situation because `OrderedDictionary` does not support generic types and losing that type safety make our code look ugly and a little unreliable.

I would love your thoughts on how we could solve the problem above or probably a new name for this type.

Reply:
How about constructing the clients in FailOverSupportedConfigurationClient. Instead of passing this type, pass a type that holds the args needed to construct the clients.

Then when the FailOverSupportedClient creates the inner clients it can keep track of which endpoint is associated with which client.